src/System.Runtime.Extensions/src/System/AppDomain.cs

```diff
@@ -26,5 +27,187 @@ public partial class AppDomain
                 AppContext.UnhandledException -= value;  
             }  
         }  
+
+        public string DynamicDirectory => null;
+        public string FriendlyName
+        {
+            get
+            {
+                Assembly assembly = Assembly.GetEntryAssembly();
+                string name = null;
+                if(assembly == null)
+                    name = "DefaultDomain";
+                else
+                    name = assembly.GetName().Name;
+                return name;             
+            }
+        }
+        public int Id => 1;
+        public bool IsFullyTrusted => true;
+        public bool IsHomogenous => true;
+        public event System.EventHandler DomainUnload { add { } remove { } }
+
+        public event System.EventHandler<System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs> FirstChanceException
+        {  
+            add  
+            {  
+                AppContext.FirstChanceException += value;  
+            }  
+  
+            remove  
+            {  
+                AppContext.FirstChanceException -= value;  
+            }  
+        }  
+
+        public event System.EventHandler ProcessExit
+        {  
+            add  
+            {  
+                AppContext.ProcessExit += value;  
+            }  
+  
+            remove  
+            {  
+                AppContext.ProcessExit -= value;  
+            }  
+        }  
+
+        public string ApplyPolicy(string assemblyName)
+        {
+            if (assemblyName == null)
+                throw new ArgumentNullException("assemblyName");
+            if (assemblyName.Length == 0) // String.Empty
```

Comment:
Nit: the `// String.Empty` comment doesn't really add much value. I'd just delete it.